classes/bodyText/Repository.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/**
+ * @file classes/bodyText/Repository.php
+ *
+ * Copyright (c) 2025 Simon Fraser University
+ * Copyright (c) 2025 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Repository
+ *
+ * @brief A repository to find and manage body text files.
+ */
+
+namespace PKP\bodyText;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\publication\Publication;
+use APP\submission\Submission;
+use Exception;
+use PKP\context\Context;
+use PKP\file\TemporaryFileManager;
+use PKP\submissionFile\SubmissionFile;
+use PKP\user\User;
+
+class Repository
+{
+    /**
+     * Map the BodyTextFile to an array including body text content and dependent files
+     */
+    public function map(BodyTextFile $bodyTextFile): array
+    {
+        $fileProps = [];
+
+        if ($bodyTextFile->submissionFile) {
+            $submission = Repo::submission()->get($bodyTextFile->submissionId);
+            $schemaMap = Repo::submissionFile()->getSchemaMap($submission, []);
+
+            $fileProps = $schemaMap->map($bodyTextFile->submissionFile);
+        }
+
+        if ($bodyTextFile->bodyTextContent) {
+            $fileProps['bodyTextContent'] = $bodyTextFile->bodyTextContent;
+        }
+
+        if ($bodyTextFile->loadingContentError) {
+            $fileProps['loadingContentError'] = $bodyTextFile->loadingContentError;
+        }
+
+        return $fileProps;
+    }
+
+    /**
+     * Returns the SubmissionFile, if any, that corresponds to the body text contents of the given submission/publication
+     */
+    public function getBodyTextFile(int $publicationId, ?int $submissionId = null): BodyTextFile
+    {
+        $submissionFileQuery = Repo::submissionFile()
+            ->getCollector()
+            ->filterByFileStages([SubmissionFile::SUBMISSION_FILE_BODY_TEXT])
+            ->filterByAssoc(Application::ASSOC_TYPE_PUBLICATION, [$publicationId]);
+
+        if ($submissionId) {
+            $submissionFileQuery = $submissionFileQuery->filterBySubmissionIds([$submissionId]);
+        }
+
+        $submissionFile = $submissionFileQuery
+            ->getMany()
+            ->first();
+
+        return new BodyTextFile(
+            $publicationId,
+            $submissionId,
+            $submissionFile,
+        );
+    }
+
+    /**
+     * Create or update the body text file for a publication
+     */
+    public function setBodyText(
+        string $bodyText,
+        int $publicationId,
+        ?int $submissionId = null,
+    ): BodyTextFile {
+        $publication = Repo::publication()->get($publicationId, $submissionId);
```

Comment:
Do we need to be passing around the `$submissionId` here since it could be null, and it should always be available form the `$publication`. It looks like, in this method at least, this is the only place the original argument `$submissionId` is used, but shouldn't the `$publicationId` be unique enough to retrieve the correct publication and not need the `$submissionId` to be passed in?

Reply:
As a more general comment on this approach throughout this PR and this class, do we need to be passing in an optionally null `$submissionId` rather than getting it from the required publication? It could still be set in constructors, etc. so it's always available, but it seems like there could be room for error in the future, but maybe I'm missing why the `$submissionId` is being passed around separately and is optional.